scipy/stats/stats.py

```diff
@@ -408,7 +416,14 @@ def gmean(a, axis=0, dtype=None):
             log_a = np.log(np.asarray(a, dtype=dtype))
     else:
         log_a = np.log(a)
-    return np.exp(log_a.mean(axis=axis))
+
+    if weights is not None:
```

Comment:
I just learned about np.asanyarray. i think that would do the trick here, actually. No need to change it @maxfrei750; i can.

Reply:
Feel free to aply changes as necessary.